source/material_model/diffusion_dislocation.cc

```diff
@@ -518,19 +541,19 @@ namespace aspect
                                    " Compositional fields can each be assigned individual"
                                    " activation energies, reference densities, thermal expansivities,"
                                    " and stress exponents. The effective viscosity is defined as"
-                                   "\n\n"
-                                   " \\[v_\\text{eff} = \\left(\\frac{1}{v_\\text{eff}^\\text{diff}}+\\frac{1}{v_\\text{eff}^\\text{dis}\\right)^{-1}\\]"
+                                   " \n\n"
+                                   " \\[v_\\text{eff} = \\left(\\frac{1}{v_\\text{eff}^\\text{diff}}+"
+                                   " \\frac{1}{v_\\text{eff}^\\text{dis}\\right)^{-1}\\]"
                                    " where"
-                                   " \\[v_\\text{eff}^\\text{diff} = A_\\text{diff}^{-1} \\exp\\left(\frac{E_\\text{diff} + PV_\\text{diff}}{RT}\\right)\\]"
-                                   " \\[v_\\text{eff}^\\text{dis} =  A_\\text{dis}^{\\frac{-1}{n_{dis}}} \\dot{\\varepsilon}^{\frac{1-n}{n}} "
-                                   "                                 \\exp\\left(\frac{E_\\text{diff} + PV_\\text{diff}}{n_\\text{dis}RT}\\right)\\]"
-                                   "\n\n"
-                                   " where $\\dot{\\varepsilon}$ is the second invariant of the strain rate tensor,"
-                                   " $A_i$ are prefactors where $i$ corresponds to diffusion or dislocation creep,"
+                                   " \\[v_\\text{i} = 0.5 * A^{-\\frac{1}{n_i}} d^\\frac{m_i}{n_i}"
+                                   " \\dot{\\varepsilon}^{\\frac{1-n_i}{n_i}}"
+                                   " \\exp\\left(\\frac{E_i^* + PV_i^*}{n_iRT}\\right)\\]"
+                                   " \n\n"
+                                   " where $d$ is grain size, $i$ corresponds to diffusion or dislocation creep,"
+                                   " $\\dot{\\varepsilon}$ is the second invariant of the strain rate tensor,"
+                                   " $R$ is the gas constant, $T$ is temperature, and $P$ is pressure."
+                                   " $A_i$ are prefactors, $n_i$ and $m_i$ are stress and grain size exponents"
                                    " $E_i$ are the activation energies, $V_i$ are the activation volumes,"
-                                   " $n_dislocation$ are stress exponents for dislocation creep,"
-                                   " $\\rho_m$ is the mantle density, $R$ is the gas constant,"
-                                   " $T$ is temperature, and $P$ is pressure."
                                    " \n\n"
                                    " The value for the components of this formula and additional"
                                    " parameters are read from the parameter file in subsection"
```

Comment:
The documentation does not talk about any iteration, but you have parameters that seem to imply that you iterate some sort of nonlinearity. Can you add a paragraph that explains this procedure?